### PR TITLE
Admin search functionality

### PR DESCRIPTION
**Pull request by dhakim87**
Added admin ability to search accounts by email and search kits by supplied kit id.  Added kit information when searching for barcodes for #104

### CODE REVIEW

File: microsetta_private_api/repo/admin_repo.py [hunk 1]
```diff
@@ -74,4 +83,63 @@ def retrieve_diagnostics_by_barcode(self, sample_barcode):
...
+            'sample_diagnostic_info': sample_assoc
+        }
+
+        return diagnostic
+
+    def retrieve_diagnostics_by_email(self, email):
+
+        # TODO: I leaned toward making this a way to discover what account to
+        #  look into rather than giving information about all accounts with
+        #  similar emails.  Does this approach work?  -DH
```
Comment (wasade) [inline microsetta_private_api/repo/admin_repo.py:136]: :+1:

File: microsetta_private_api/api/microsetta_private_api.yaml [hunk 2]
```diff
@@ -834,6 +834,57 @@ paths:
           $ref: '#/components/responses/401Unauthorized'
         '404':
           $ref: '#/components/responses/404NotFound'
+  '/admin/search/kit/{kit_id}':
+    get:
+      operationId: microsetta_private_api.admin.admin_impl.search_kit_id
+      tags:
+        - Admin
+      summary: Retrieve diagnostic information about a sample by barcode
```
Comment (wasade) [inline microsetta_private_api/api/microsetta_private_api.yaml:842]: Copypasta, this is by kit id :)

File: microsetta_private_api/api/microsetta_private_api.yaml [hunk 3]
```diff
@@ -834,6 +834,57 @@ paths:
...
+          schema:
+            $ref: '#/components/schemas/kit_name'
+          required: true
+      responses:
+        '200':
+          description: Object containing linked account (if any), source (if any), sample (if any) and information about the barcode
+          content:
+            application/json:
+              schema:
+                type: object
```
Comment (wasade) [inline microsetta_private_api/api/microsetta_private_api.yaml:857]: Goal is to define these later right? If so, possible to create an issue?

Comment (dhakim87) [reply microsetta_private_api/api/microsetta_private_api.yaml:857]: I'm torn on this.  If the admin user interface is parsing a fixed json schema, then yes we need to define what it is.  If that interface is just displaying general json, then I can just return whatever debugging information I have.  

Defining these schemas in yaml is semi painful up front, and will entail a maintenance cost in keeping the models and any to_api adaptation up to date when our objects are changed.  

For the various GET operations in the admin interface, I don't see a particular benefit to fully specifying the schema.  There is the option of a partial schema object in the yaml, guaranteeing the existence of several fixed fields and leaving any additional information open.  For the POSTs which must write exact information to the table, I think we need full specification.

Comment (wasade) [reply microsetta_private_api/api/microsetta_private_api.yaml:857]: The admin interface will need to make at least some expectations on this structure. If they chance, then we may silently break the admin interface. So I think we incur pain here or there. Ultimately, the royal we are maintaining this, so if you think there is less pain in manipulating jinja, then lets proceed as is

Comment (dhakim87) [reply microsetta_private_api/api/microsetta_private_api.yaml:858]: Created #111

**Pull request merged**